README.rst
Python Client for Channel Services API
======================================

|stable| |pypi| |versions|

`Channel Services API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-channel.svg
   :target: https://pypi.org/project/google-cloud-channel/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-channel.svg
   :target: https://pypi.org/project/google-cloud-channel/
.. _Channel Services API: https://cloud.google.com/channel/
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/cloudchannel/latest
.. _Product Documentation:  https://cloud.google.com/channel/

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Channel Services API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Channel Services API.:  https://cloud.google.com/channel/
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-channel


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-channel

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Channel Services API
   to see other available methods on the client.
-  Read the `Channel Services API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Channel Services API Product documentation:  https://cloud.google.com/channel/
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

### [1.2.2](https://www.github.com/googleapis/python-channel/compare/v1.2.1...v1.2.2) (2021-07-27)


### Features

* add Samples section to CONTRIBUTING.rst ([#72](https://www.github.com/googleapis/python-channel/issues/72)) ([b70a090](https://www.github.com/googleapis/python-channel/commit/b70a0900ab7b820e623a286cbd6f0e5c29ad9256))


### Bug Fixes

* enable self signed jwt for grpc ([#76](https://www.github.com/googleapis/python-channel/issues/76)) ([016111e](https://www.github.com/googleapis/python-channel/commit/016111ee4750d047c44324bf3dca752560840376))


### Miscellaneous Chores

* release as 1.2.2 ([#77](https://www.github.com/googleapis/python-channel/issues/77)) ([2e950a4](https://www.github.com/googleapis/python-channel/commit/2e950a41e5099facad90b884fe24f43e00b59255))

### [1.2.1](https://www.github.com/googleapis/python-channel/compare/v1.2.0...v1.2.1) (2021-07-20)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#71](https://www.github.com/googleapis/python-channel/issues/71)) ([36ba3c3](https://www.github.com/googleapis/python-channel/commit/36ba3c36d9ffb37c1f156fd69f7216c331f55f87))

## [1.2.0](https://www.github.com/googleapis/python-channel/compare/v1.1.1...v1.2.0) (2021-07-01)


### Features

* add always_use_jwt_access ([#64](https://www.github.com/googleapis/python-channel/issues/64)) ([45621e5](https://www.github.com/googleapis/python-channel/commit/45621e5eabad6ff6979b89d619dcbf979ed78d63))


### Bug Fixes

* disable always_use_jwt_access ([2f8fa52](https://www.github.com/googleapis/python-channel/commit/2f8fa52e232e5468e6797d2b14166c7f319a0ca1))
* disable always_use_jwt_access ([#68](https://www.github.com/googleapis/python-channel/issues/68)) ([2f8fa52](https://www.github.com/googleapis/python-channel/commit/2f8fa52e232e5468e6797d2b14166c7f319a0ca1))


### Documentation

* omit mention of Python 2.7 in 'CONTRIBUTING.rst' ([#1127](https://www.github.com/googleapis/python-channel/issues/1127)) ([#59](https://www.github.com/googleapis/python-channel/issues/59)) ([f358de3](https://www.github.com/googleapis/python-channel/commit/f358de3eb94fc9c870a899e3d94d93c5f14b5b0d)), closes [#1126](https://www.github.com/googleapis/python-channel/issues/1126)

### [1.1.1](https://www.github.com/googleapis/python-channel/compare/v1.1.0...v1.1.1) (2021-06-16)


### Bug Fixes

* exclude docs and tests from package ([#55](https://www.github.com/googleapis/python-channel/issues/55)) ([7747953](https://www.github.com/googleapis/python-channel/commit/7747953eb02ae952108cc9d2dcd66f939e60e115))

## [1.1.0](https://www.github.com/googleapis/python-channel/compare/v1.0.0...v1.1.0) (2021-06-10)


### Features

* Add a new enum value LICENSE_CAP_CHANGED to enum EntitlementEvent.Type. ([164539d](https://www.github.com/googleapis/python-channel/commit/164539dc0892481a739bfe4372c35be8d74480d9))
* Add a new LookupOffer RPC and LookupOfferRequest proto. ([164539d](https://www.github.com/googleapis/python-channel/commit/164539dc0892481a739bfe4372c35be8d74480d9))
* Add additional_bindings to HTTP annotations of Customer related APIs (list/create/get/update/delete). ([164539d](https://www.github.com/googleapis/python-channel/commit/164539dc0892481a739bfe4372c35be8d74480d9))
* Add/Update API definitions for Cloud Channel API ([#53](https://www.github.com/googleapis/python-channel/issues/53)) ([164539d](https://www.github.com/googleapis/python-channel/commit/164539dc0892481a739bfe4372c35be8d74480d9))
* Update descriptions of APIs. ([164539d](https://www.github.com/googleapis/python-channel/commit/164539dc0892481a739bfe4372c35be8d74480d9))

## [1.0.0](https://www.github.com/googleapis/python-channel/compare/v0.3.0...v1.0.0) (2021-06-02)


### Features

* bump release level to production/stable ([#46](https://www.github.com/googleapis/python-channel/issues/46)) ([3460f68](https://www.github.com/googleapis/python-channel/commit/3460f683a4e8f1a2f94a1eeb92af7eae1dcdb02b))


### Miscellaneous Chores

* release as 1.0.0 ([#51](https://www.github.com/googleapis/python-channel/issues/51)) ([00710b6](https://www.github.com/googleapis/python-channel/commit/00710b6566540ba111c5bb3705f892e8f743b1c2))

## [0.3.0](https://www.github.com/googleapis/python-channel/compare/v0.2.1...v0.3.0) (2021-05-28)


### Features

* support self-signed JWT flow for service accounts ([c218e2e](https://www.github.com/googleapis/python-channel/commit/c218e2efeef2f2e4cf1646644a8d4c38be021bdf))


### Bug Fixes

* add async client to %name_%version/init.py ([c218e2e](https://www.github.com/googleapis/python-channel/commit/c218e2efeef2f2e4cf1646644a8d4c38be021bdf))
* **deps:** add packaging requirement ([#43](https://www.github.com/googleapis/python-channel/issues/43)) ([e368062](https://www.github.com/googleapis/python-channel/commit/e36806211b8b3392811eb5ee1047517f840265c7))

### [0.2.1](https://www.github.com/googleapis/python-channel/compare/v0.2.0...v0.2.1) (2021-04-07)


### Bug Fixes

* BREAKING deprecate TransferableSkus fields ([#14](https://www.github.com/googleapis/python-channel/issues/14)) ([0d3b493](https://www.github.com/googleapis/python-channel/commit/0d3b4939cdae196ea9b0edc00e13f61d7d71777d))

## [0.2.0](https://www.github.com/googleapis/python-channel/compare/v0.1.0...v0.2.0) (2021-02-11)


### Features

* add Pub/Sub endpoints for Cloud Channnel API ([#9](https://www.github.com/googleapis/python-channel/issues/9)) ([2c483c8](https://www.github.com/googleapis/python-channel/commit/2c483c8ec24bba25fdea7a1f46d3d5396fe2076a))

## 0.1.0 (2021-01-14)


### Features

* generate v1 ([a95c9cf](https://www.github.com/googleapis/python-channel/commit/a95c9cf86cc9188c1e3eb8535c62367d141658cc))


owl-bot-staging/v1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")


nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
]

@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/channel_v1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python='3.7')
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=['3.6', '3.7'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python='3.6')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
